File: src/main/java/al132/alchemistry/recipes/AtomizerRecipe.java
package al132.alchemistry.recipes;

import net.minecraft.item.ItemStack;
import net.minecraftforge.common.util.LazyOptional;
import net.minecraftforge.fluids.FluidStack;

public class AtomizerRecipe {

    public FluidStack input;
    public ItemStack output;
    public boolean reversible;

    public AtomizerRecipe(FluidStack input, ItemStack output) {
        this(false, input, output);
    }

    public AtomizerRecipe(boolean reversible, FluidStack input, ItemStack output) {
        this.reversible = reversible;
        this.input = input;
        this.output = output;
    }
}

File: src/main/kotlin/al132/alchemistry/recipes/AtomizerRecipe.kt
package al132.alchemistry.recipes

import net.minecraft.item.ItemStack
import net.minecraftforge.fluids.Fluid
import net.minecraftforge.fluids.FluidStack

/**
 * Created by al132 on 4/29/2017.
 */


data class AtomizerRecipe(private val inputFluid: FluidStack, private val outputOne: ItemStack) {

    constructor(fluid: Fluid, fluidQuantity: Int, output: ItemStack) : this(FluidStack(fluid, fluidQuantity), output)

    val input: FluidStack
        get() = this.inputFluid.copy()

    val output: ItemStack
        get():ItemStack = outputOne.copy()
}